src/dmd/cparse.d

```diff
@@ -1411,6 +1411,52 @@ final class CParser(AST) : Parser!AST
         Specifier specifier;
         auto tspec = cparseDeclarationSpecifiers(level, specifier);
 
+        /* If a declarator does not follow, it is unnamed
+         */
+        if (token.value == TOK.semicolon &&
+            tspec && tspec.isTypeTag())
+        {
```

Comment:
Forgot to consume the semicolon?

Reply:
yup

Reply:
OK, perhaps I should add an assert at this part of cparse, because that is the only way that we'll capture these subtle things. :-)
```diff
@@ -1429,6 +1476,11 @@ final class CParser(AST) : Parser!AST
                 dt = cparseDeclarator(tspec, id);
             if (!dt)
             {
+                if (!global.errors)
+                {
+                    error("panic without errors");
+                    global.errors = 0;
+                }
                 panic();
                 nextToken();
                 break;          // error recovery
```